### PR TITLE
routed grpc through langgraph

### PR DESCRIPTION
**Pull request by LilliDarling**
## Summary by Sourcery

Route gRPC voice streaming through the shared LangGraph-based orchestrator, unify MQTT and gRPC AI pipelines, and harden server lifecycle, signal handling, and test tooling.

New Features:
- Introduce a shared NAILAOrchestrator entry point that both MQTT and gRPC use for LangGraph execution, conversation memory, and AI services.
- Add transport-aware audio delivery callbacks so the orchestration graph can stream TTS audio over gRPC while continuing to publish via MQTT.
- Provide CLI test clients for gRPC StreamConversation and MQTT orchestration topics to exercise the end-to-end AI pipeline.

Bug Fixes:
- Prevent audio buffer overflows in the gRPC service by enforcing a maximum buffered PCM size per utterance.
- Fix potential shutdown races by switching signal handling to an asyncio.Event model and ensuring the gRPC server fully terminates on stop.
- Ensure MQTT message handling from the paho network thread safely schedules work onto the asyncio event loop.

Enhancements:
- Refactor the gRPC servicer to delegate post-STT logic to the orchestrator instead of directly calling LLM and TTS services, including better error handling and text-only fallbacks.
- Extend the orchestrator and LangGraph orchestration graph to accept a config object for transport-specific callbacks, and forward it into the response generator.
- Improve response generation behavior to synthesize audio conditionally based on transport and input type, and make LLM timeout handling safer by avoiding retries that can crash the model.
- Tune default LLM context and token limits and GPU offload heuristics for better latency and VRAM usage.
- Simplify vision processing memory usage by clearing image data from graph state after processing.
- Adjust MQTT AI error topics, health metrics, and protocol handler wiring to work cleanly with the shared orchestrator and updated stats APIs.
- Streamline cross-platform signal handling, lifecycle shutdown semantics, and MQTT client API usage to match newer library requirements.

Documentation:
- Update README with clearer, platform-specific instructions for enabling GPU acceleration for llama-cpp and optional CUDA-optimized PyTorch.

Tests:
- Expand gRPC service tests to cover orchestrator delegation, transcription propagation, empty/failed TTS behavior, and STT edge cases like empty transcriptions.
- Extend orchestration graph tests to verify config passthrough into nodes and response generator, and adjust existing tests for the new config parameter.

### CODE REVIEW

File: ai-server/rpc/service.py [hunk 1]
```diff
@@ -277,107 +300,76 @@ async def _process_utterance(
...
-        logger.info(
-            "llm_result",
-            device_id=device_id,
-            response_length=len(response_text),
-        )
-
-        if context and not context.is_active():
-            logger.info("client_disconnected_after_llm", device_id=device_id)
-            return
+        async def _grpc_audio_delivery(audio_data, text: str, is_final: bool = True):
```
[the remark this answers is not in the document]
Comment (Rhovian) [reply ai-server/rpc/service.py:335]: @LilliDarling I don't think this is a blocker for this PR to land, but worth making an issue out of it for future optimization work

File: ai-server/rpc/service.py [hunk 2]
```diff
@@ -102,6 +115,14 @@ async def _read_inputs():
 
                     elif event == naila_pb2.SPEECH_EVENT_CONTINUE:
                         if pcm_data:
+                            if len(audio_buffer) + len(pcm_data) > MAX_AUDIO_BUFFER_BYTES:
+                                logger.warning(
+                                    "audio_buffer_overflow",
+                                    device_id=device_id,
+                                    buffer_size=len(audio_buffer),
+                                )
+                                audio_buffer.clear()
```
[the remark this answers is not in the document]
Comment (Rhovian) [reply ai-server/rpc/service.py:136]: @LilliDarling emitting an error here makes sense to do for this pr

File: ai-server/utils/platform.py [hunk 3]
```diff
@@ -11,57 +11,48 @@
...
-                logger.debug("signal_handlers_configured", platform="unix", signals="SIGINT, SIGTERM, SIGHUP")
-
+            signals = [signal.SIGINT, signal.SIGTERM]
+            if hasattr(signal, 'SIGHUP'):
+                signals.append(signal.SIGHUP)
+            for sig in signals:
+                signal.signal(sig, self._signal_handler)
+            logger.debug("signal_handlers_configured", signals=[s.name for s in signals])
         except Exception as e:
             logger.warning("signal_handler_setup_failed", error=str(e), error_type=type(e).__name__)
```
[the remark this answers is not in the document]
Comment (Rhovian) [reply ai-server/utils/platform.py]: also worth the small fix

**Review by Rhovian — CHANGES_REQUESTED** [review]
